src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildLogRegexMacro.java

```diff
@@ -234,15 +244,72 @@ String getContent(BufferedReader reader)
                 }
             }
             if (numLinesTruncated > 0) {
-                insidePre = stopPre(buffer, insidePre);
-                appendLinesTruncated(buffer, numLinesTruncated, asHtml);
+                preStartLoc = stopPre(matchResults, preStartLoc, preRanges);
+                appendLinesTruncated(matchResults, numLinesTruncated, asHtml);
             }
         }
-        insidePre = stopPre(buffer, insidePre);
-        if (buffer.length() == 0) {
+        preStartLoc = stopPre(matchResults, preStartLoc, preRanges);
+        if (matchResults.size() == 0) {
             return defaultValue;
         }
-        return buffer.toString();
+        if (maxTailMatches > 0 && matchResults.size() > maxTailMatches) {
+            int tailStartLocation;
+            if (asHtml) {

```

Comment:
@slide I was considering to handle this like how the truncated lines are handled in the else case, but this will become unreliable if there are pre blocks in the log output itself, so decided to not do it. Please let me know if you think otherwise.